MekHQ/src/mekhq/campaign/parts/MissingOmniPod.java

```diff
@@ -0,0 +1,231 @@
+                if (null == wn2.getAttributes().getNamedItem("type")) {
+                    MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
+                            "OmniPod lacks part type attribute."); //$NON-NLS-1$
+                } else if (null == wn2.getAttributes().getNamedItem("tonnage")) {
+                    MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
+                            "OmniPod lacks partType tonnage attribute."); //$NON-NLS-1$
+                } else {
+                    String type = wn2.getAttributes().getNamedItem("type").getTextContent();
+                    int tonnage = Integer.parseInt(wn2.getAttributes().getNamedItem("tonnage").getTextContent());
+                    if (type.equals("AeroHeatSink")) {
+                        int hsType = -1;
+                        if (null != wn2.getAttributes().getNamedItem("hsType")) {
+                            hsType = Integer.parseInt(wn2.getAttributes().getNamedItem("hsType").getTextContent());
+                        }
+                        if (hsType != Aero.HEAT_SINGLE && hsType != Aero.HEAT_DOUBLE) {
+                            MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
+                                    "Aero heatsink OmniPod does not have a legal value for heat sink type; using SINGLE"); //$NON-NLS-1$
+                            hsType = Aero.HEAT_SINGLE;
+                        }
+                        partType = new AeroHeatSink(0, hsType, false, campaign);
+                    } else {
+                        EquipmentType et = EquipmentType.get(type);
+                        if (null == et) {
+                            MekHQ.getLogger().log(getClass(), METHOD_NAME, LogLevel.ERROR,
+                                    "Unknown part type " + type + " for OmniPod"); //$NON-NLS-1$
+                            //Throw a generic value in there to prevent NPE but still indicate a problem
+                            et = EquipmentType.get(EquipmentType
+                                    .getStructureTypeName(EquipmentType.T_STRUCTURE_STANDARD));
+                        }
+                        if (et instanceof MiscType
+                                && (et.hasFlag(MiscType.F_HEAT_SINK)
+                                        || et.hasFlag(MiscType.F_DOUBLE_HEAT_SINK)
+                                        || et.hasFlag(MiscType.F_IS_DOUBLE_HEAT_SINK_PROTOTYPE))) {
+                            partType = new HeatSink(0, et, -1, false, campaign);
+                        } else if (et instanceof MiscType && et.hasFlag(MiscType.F_JUMP_JET)) {
+                            partType = new JumpJet(tonnage, et, -1, false, campaign);
+                        } else if (et instanceof MiscType
+                                && et.hasFlag(MiscType.F_MASC)
+                                && (et.getSubType() & MiscType.S_SUPERCHARGER) == 0) {
```

Comment:
Per TM, MASC can't be placed in an omnipod.  It looks like a Supercharger can be.  We should probably also allow other variable weight equipment to write to / load from XML, like hatchets, targeting computers and such. I don't see anything that says those items can't be omnipods.

Reply:
@mkerensky Re-read that check, it **is** checking for a Supercharger. Code defines that as a subtype of MASC. TCs can definitely be pod mounted, almost all the Clan ones are. I don't know about physical weapons, but maybe.

Reply:
This has nothing to do with determining whether a part can be podded or not. It deals with tracking what type of part the pod is built for writing whatever data is needed to restore the `EquipmentPart` to the campaign file.